moon/cfc_chat_transit/server/init.moon

```diff
@@ -63,17 +63,55 @@ ChatTransit.ReceiveMessage = (ply, text, teamChat) =>
     irisId = "none"
 
     struct =
-        Realm: Realm
-        Content: text
-        Avatar: avatar
-        SteamName: steamName
-        SteamId: steamId
-        IrisId: irisId
+        Type: "message"
+        data:
+            Realm: Realm
+            Type: "message"
+            Content: text
+            Avatar: avatar
+            SteamName: steamName
+            SteamId: steamId
+            IrisId: irisId
 
     message = TableToJSON struct
 
     @WebSocket\write message
 
     @Logger\debug "Sent message '#{text}' to websocket"
 
+ChatTransit.PlayerConnected = (data) =>
+    steamName = data.name
+    steamId = data.networkid
+
+    struct =
+        Type: "connect"
+        data:
+            Realm: Realm
+            SteamName: steamName
+            SteamId: steamId
+
+    message = TableToJSON struct
+
+    @WebSocket\write message
+
+ChatTransit.PlayerDisconnected = (ply) =>
+    steamName = data.name
+    steamId = data.networkid
+
+    struct =
+        Type: "disconnect"
+        data:
```

Comment:
this needs to be `Data` because your struct in go has the field `Data`
this applies to the  other messages as well

Reply:
Applied